### ISSUE TITLE
Finishing-Upgrade (STATE_CHANGED)

### ISSUE THREAD

**Issue by sankaet** [rancher/rancher#7144]
Hi, when I am updating containers I am getting the following

<img width="992" alt="screen shot 2016-12-20 at 7 21 16 am" src="https://cloud.githubusercontent.com/assets/1951158/21355989/eeed9ad2-c684-11e6-9980-3f561611640c.png">

After we updated to 1.2.1 strange errors like this are happening

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.2.1` Cattle: `v0.174.8` UI: `v1.2.39` |
|Access|`github` `admin`|
|Orchestration|`Cattle`|
|Route|`service.log`|

**Comment by deniseschannon** [comment]
@sankaet Can you provide more information on your previous service and what you changed to upgrade? 

Is there a docker-compose.yml and rancher-compose.yml file that you can send? If you click on the stack name and export the config that will show you the services in your stack.

**Comment by bigfoot90** [comment]
@deniseschannon I have this issue also. It happens when you set a health check (in my case I check the http 200 status code) for a service.

**Comment by sankaet** [comment]
Hi @deniseschannon, @bigfoot90 is correct this was a function of health checks, if you are not getting a 200 response on the health check call, it will give the `STATE_CHANGED` error.

**Comment by Shemeikka** [comment]
I have this issue also.

Logs from rancher-server docker container (docker logs -f rancher-server)

    2016-12-30 07:13:07,427 ERROR [dff6a94e-d6d4-4244-aeb9-baed6d7cf123:3604] [service:24]   [service.finishupgrade] [] [ecutorService-8] [c.p.e.p.i.DefaultProcessInstanceImpl] Exiting with code [STATE_CHANGED] : STATE_CHANGED 
    2016-12-30 07:13:07,433 ERROR [:] [] [] [] [ecutorService-8] [.e.s.i.ProcessInstanceDispatcherImpl]   Process [service.finishupgrade:3604] on [24] failed, exit [STATE_CHANGED] :  io.cattle.platform.engine.process.impl.ProcessExecutionExitException: STATE_CHANGED
    
I have a web (Java tomcat7) container in a stack. I upgraded this container and added HTTP/1.0 GET health check to route / and clicked Upgrade.

Upgrade works to a point but when I click Finish upgrade, I immediately get this error and container doesn't respond to HTTP requests. Only option is to cancel the upgrade and rollback.

Timeouts on the health check doesn't seem to matter.

I'm using 

    Rancher	v1.2.2
    Cattle	v0.174.13
    User Interface	v1.2.40
    Rancher Compose	v0.12.1
    Database  PostgreSQL 9.6 (AWS RDS)

**Comment by flypenguin** [comment]
I had the same error right now. I think your error messages are *really* bad in that case - cause I needed to google this issue to find out that my health check wasnt working properly. 

STATE_CHANGED is not what I connect to this root cause.

**Comment by sankaet** [comment]
Actually updating to [v1.2.2](https://github.com/rancher/rancher/releases/tag/v1.2.2) resolves these issues

**Comment by alena1108** [comment]
@sangeethah would be a bit tricky to test. The error can happen during finishUpgrade for services with health check enabled. If we do have this test in our validation, just watch it. If not, can be tested manually.

**Comment by galal-hussein** [comment]
@alena1108 I validated the fix using the following steps:

- Run Rancher and register few agents.
- Deploy wordpress catalog from the community catalog.
- Upgrade the wordpress service and add healthcheck for port 80.
- Click finish upgrade after the container is available.

I compared between doing the same steps on 1.4.0-rc3 and 1.2.1 i saw the `Finishing-Upgrade (STATE_CHANGED)` on 1.2.1 only and i can't see this problem anymore on 1.4.0-rc3
